xgfield/fieldsky.py

```diff
@@ -27,30 +31,40 @@ class FieldSky:
     def __init__(self, **kwargs):
 
         import xgfield.defaults as fd
+
         self.ID    = kwargs.get(    'ID',fd.ID)
         self.N     = kwargs.get(     'N',fd.N)
         self.Lbox  = kwargs.get(  'Lbox',fd.Lbox)
         self.Nside = kwargs.get( 'Nside',fd.Nside)
         self.input = kwargs.get( 'input',fd.input)
-        self.nproc = kwargs.get( 'nproc',1)
-        self.rank  = kwargs.get(  'rank',0)
+        self.gpu   = kwargs.get(   'gpu',fd.gpu)
+        self.mpi   = kwargs.get(   'mpi',fd.mpi)
+
+        self.nproc  = kwargs.get( 'nproc',1)
+        self.rank   = kwargs.get(  'rank',0)
+        self.cube   = kwargs.get(  'cube')
 
-        self.cube  = kwargs.get(  'cube')
         if self.cube is not None:
             self.input = 'cube'
             self.N    = self.cube.N
             self.Lbox = self.cube.Lbox
 
-        if self.input == 'websky':
-            self.displacements = _get_websky_displacements()
-        elif self.input == 'cube':
+        if self.input == 'cube':
+            cube = self.cube
             self.displacements = {}
             self.displacements['type']  = 'arraylist'
             self.displacements['data']  = [cube.s1x,cube.s1y,cube.s1z,cube.s2x,cube.s2y,cube.s2z]
-            self.displacements['start'] = (     0,self.N//nproc*rank,         0)
-            self.displacements['end']   = (self.N,self.N//nproc*(rank+1),self.N)
-
-         return
+            # store the location of local cube data within the global cube here, using the MPI rank information
+            # currently set here to y-direction sharding as is the case for sharded exgalsky FFTs, assuming
+            # one GPU per MPI rank
+            self.displacements['start'] = {}
+            self.displacements['stop']  = {}
+            self.displacements['start']['x'] = 0
+            self.displacements['stop']['x']  = self.N
+            self.displacements['start']['y'] = self.N//self.nproc*self.rank
```

Comment:
This is probably fine for now. But we should be setting the start and stop from the `MPI_backend` without exposing the MPI information in `fieldsky`. I suggest we add this to issues we are tracking.

Reply:
@1cosmologist this sounds good to me, please feel free to add this as an issue.